### PR TITLE
feat(i18n): Allow localizing menus with i18n dynamically

### PR DESCRIPTION
**Pull request by CcydtN**
Adding feature described in this link:
[https://gohugo.io/content-management/multilingual/#dynamically-localizing-menus-with-i18n](https://gohugo.io/content-management/multilingual/#dynamically-localizing-menus-with-i18n)

I figure out two issue when adding `i18n/[language].toml` to the root directory like the link describe. 
First, the text in sidebar is not changing according to the language file. Even thought "identifier" is added in every element of the menu.
Second, clicking the sidebar button after changing language will redirect the page to its default language.

So I modify the corresponding html to solve this two problem.

### CODE REVIEW

**Comment by CaiJimmy** [comment]
Thank you so much for submitting your pull request!

Sorry for the late reply. Unfortunately, I'm still tied up with academic tasks and won't be able to review pull requests for the time being.

Once I have more time, I'll take a look at your pull request and provide feedback.

Thank you again for your contributions!

File: layouts/partials/sidebar/left.html [hunk 1]
```diff
@@ -62,15 +62,15 @@ <h2 class="site-description">{{ .Site.Params.sidebar.subtitle }}</h2>
         {{ range .Site.Menus.main }}
         {{ $active := or (eq $currentPage.Title .Name) (or ($currentPage.HasMenuCurrent "main" .) ($currentPage.IsMenuCurrent "main" .)) }}
         <li {{ if $active }} class='current' {{ end }}>
-            <a href='{{ .URL }}' {{ if eq .Params.newTab true }}target="_blank"{{ end }}>
+            <a href='{{ .URL | absLangURL }}' {{ if eq .Params.newTab true }}target="_blank"{{ end }}>
```
Comment (CaiJimmy) [inline layouts/partials/sidebar/left.html:65]: `absLangURL` would create problem when the menu item is added through page's frontmatter. See https://github.com/CaiJimmy/hugo-theme-stack/issues/770#issuecomment-1678713221

**Review by CaiJimmy — CHANGES_REQUESTED** [review]
Thanks for the PR and sorry for the late reply!

File: layouts/partials/sidebar/left.html (hunk 1 above)
Comment (CcydtN) [reply layouts/partials/sidebar/left.html:65]: I see...
I couldn't think of another solution at the moment. Let me close the MR for now and think about it later.
Anyway, thanks for review.

Comment (CaiJimmy) [reply layouts/partials/sidebar/left.html:65]: Thanks to you! 

And don't worry about it! It's still possible to localize the menu manually in the config file.

**Pull request closed without merging**